tools/autograd/gen_variable_type.py

```diff
@@ -165,7 +165,7 @@
     'neg', 'complex', 'select', '_s_where', 'as_strided', 'slice', 'constant_pad_nd',
     'unbind', 'split', 'split_with_sizes', 'unsafe_split', 'split_with_sizes_backward',
     'dot', 'vdot', 'cholesky', 'triangular_solve', 'mm', '_unsafe_view', 'mv', 'ger',
-    'bmm', 'diagonal'
+    'bmm', 'diagonal', 'nonzero'
```

Comment:
should this be a different PR?  what does this have to do with the other changes?

Reply:
yeah it is an unrelated change that's true